src/nvim/ui_compositor.c

```diff
@@ -101,6 +101,21 @@ bool ui_comp_should_draw(void)
   return composed_uis != 0 && valid_screen;
 }
 
+static int compare_zindex(const void *a, const void *b)
+{
+  int za = (*(ScreenGrid **)a)->zindex;
+  int zb = (*(ScreenGrid **)b)->zindex;
+  return za == zb ? 0 : (za < zb ? -1 : 1);
```

Comment:
The sorting is not stable. If two layers have the same zindex, their order should be kept, so compare `comp_index` in that case:
```suggestion
  size_t ia = (*(ScreenGrid **)a)->comp_index;
  size_t ib = (*(ScreenGrid **)b)->comp_index;
  return za == zb ? (ia == ib ? 0 : (ia < ib ? -1 : 1)) : (za < zb ? -1 : 1);
```

Reply:
sound reasonable..